src/frontend/src/states/ApiState.tsx

```diff
@@ -194,7 +192,7 @@ export function apiEndpoint(path: ApiPaths): string {
 /**
  * Construct an API URL with an endpoint and (optional) pk value
  */
-export function apiUrl(path: ApiPaths, pk?: any): string {
+export function apiUrl(path: ApiPaths, pk?: any, data?: any): string {
```

Comment:
Why not type this as
```suggestion
export function apiUrl(path: ApiPaths, pk?: any, data?: Record<string, string | number>): string {
```
(Then the replacement value would need to be casted to a string using an f string, but that way we could also hand in some ids as numbers.)

Reply:
Tbh my understanding of the typing system is still entry level